### ISSUE TITLE
AWS::StepFunctions::StateMachine fails when using old definition style

### ISSUE THREAD

**Issue by ccassidy-vaisala** [aws-cloudformation/cfn-lint#4196]
### CloudFormation Lint Version

1.38.0

### What operating system are you using?

Amazon Linux 2023

### Describe the bug

It appears #3524 may have cropped up again, I have a step function using the legacy Iterator/ItemsPath configuration and it is throwing E3601:

```
E3601 'ItemProcessor' is a required property
template.yml:257:11

E3601 Additional properties are not allowed ('Iterator' was unexpected)
template.yml:270:13
```

### Expected behavior

Do not throw error, maybe warning considering this legacy style is deprecated?

### Reproduction template

```
Resources:
  StateMachine:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      Definition:
        StartAt: ProcessMap
        States:
          ProcessMap:
            Type: Map
            Next: EndState
            ItemsPath: "$.items"
            Iterator:
              StartAt: ProcessingTask
              States:
                ProcessingTask:
                  Type: Task
                  End: true
          EndState:
            End: true
            Type: Pass
```

**Comment by kddejong** [comment]
```A Map State MUST contain an object field named "ItemProcessor" (or deprecated "Iterator") which MUST contain fields named "States" and "StartAt", whose meanings are exactly like those in the top level of a State Machine. The "ItemProcessor" field performs the same function as the "Iterator" field, which is now deprecated in the Map State.```
